src/sentry/notifications/notifications/rules.py

```diff
@@ -117,10 +118,17 @@ def get_context(self) -> MutableMapping[str, Any]:
             fallthrough_choice=self.fallthrough_choice,
         )
         fallback_params: MutableMapping[str, str] = {}
+        group_inbox_reason = GroupInbox.objects.filter(
+            group=self.group, project=self.project
+        ).first()
+        group_header = "New Alert"
+        if group_inbox_reason:
+            group_header = get_inbox_reason_text(GroupInboxReason(group_inbox_reason.reason))
```

Comment:
nit: I would pass `group_inbox_reason` directly to `get_inbox_reason_text` and let it handle the case where is not defined within it, thus, you would give all responsibility to the function.